docs/Ubuntu24 Desktop VM.md

```diff
@@ -0,0 +1,245 @@
+At the end of the file add:
+```
+PS1="${debian_chroot:+($debian_chroot)}\[\033[00;35m\]\u@\h\[\033[02;33m\]\$(git branch 2> /dev/null | sed -e '/^[^*]/d' -e 's/* \(.*\)/ \1/')\[\033[00m\]:\[\033[02;37m\]\w\[\033[00m\]$ "
+```
+Then run the following command to pick up the changes:
+```
+$ . ~/.bashrc
+```
+
+#### Install git
+_Replace `username@gmail.com` with your email address and `username` with your github username._
+```
+$ sudo apt update
+$ sudo apt install git
+$ git config --global credential.helper store
+$ git config --global user.email username@gmail.com
+$ git config --global user.name username
+```
+
+#### Install docker
+```
+$ sudo groupadd docker
+$ sudo apt-get update
+$ sudo apt-get install ca-certificates curl gnupg
+$ sudo install -m 0755 -d /etc/apt/keyrings
+$ curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+$ sudo chmod a+r /etc/apt/keyrings/docker.gpg
+$ echo \
+  "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
+  "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+$ sudo apt-get update
+$ sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+$ nano ~/.bash_aliases
+$ sudo systemctl status docker
+```
+If everything was successful, you should see "Active: active (running)" in the output.
+
+#### Checkout the code
+ If you haven't already, fork the RAWeb repository into your github space
+
+_Replace `username` with your github username._
+```
+$ mkdir ~/source
+$ cd ~/source
+$ git clone https://github.com/username/RAWeb.git
+$ cd RAWeb
+$ git remote add upstream https://github.com/RetroAchievements/RAWeb.git
+$ git fetch upstream
+$ git branch --set-upstream-to upstream/master
+$ git pull
+$ git push origin
+  - Username = (github username)
+  - Password = (github password or personal access token)
+```
+This sets your github credentials and ensures your fork is up-to-date. Because we set up a credential helper earlier, you shouldn't need to enter your credentials again unless they change.
+
+_Replace `username` with your **local** username._
+```
+$ git config --global --add safe.directory /home/username/source/RAWeb
+```
+This prevents a "dubious ownership" warning after sail installs all its stuff
+
+#### Define an alias for sail
+Open the `.bashrc` file:
+```
+$ nano .bashrc
+```
+And add the following line:
+_Replace `username` with your local username._
+```
+alias sail='sudo /home/username/source/RAWeb/vendor/bin/sail'
+```
+Reload it to pick up the change.
+```
+$ . ~/.bashrc
+```
+
+#### Setup local installation
+```
+$ cp .env.example .env
+$ nano .env
+```
+Find the commented out `DEBUGBAR_ENABLED` setting and uncomment it and set it to `"true"`
+Find the commented out `QUEUE_CONNECTION` settings and add a new one that sets `QUEUE_CONNECTION` to `"sync"`
```

Comment:
Possible typo: should "settings" be "setting"?

Reply:
Nope.
https://github.com/RetroAchievements/RAWeb/blob/9db9a76beff5ee619b014c62848c908adef34afc/.env.example#L69-L70